Create `WebPageTesting` and `WebPageProxyTesting`

charliewolfe opened the pull request:
#### cf53b45ca2ef1cc724b4993ff341891ea70e8cda
<pre>
Create `WebPageTesting` and `WebPageProxyTesting`
<a href="https://rdar.apple.com/130555180">rdar://130555180</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=275884">https://bugs.webkit.org/show_bug.cgi?id=275884</a>

Reviewed by Alex Christensen.

WebPage and WebPageProxy have many functions that are intended to only be used for testing, this change
begins to move those functions into separate files. More testing-only functions will be added as we
create replacements for injected bundle API used by WKTR.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageClearWheelEventTestMonitor):
(WKPageDumpPrivateClickMeasurement):
(WKPageClearPrivateClickMeasurement):
(WKPageSetPrivateClickMeasurementOverrideTimerForTesting):
(WKPageMarkAttributedPrivateClickMeasurementsAsExpiredForTesting):
(WKPageSetPrivateClickMeasurementEphemeralMeasurementForTesting):
(WKPageSimulatePrivateClickMeasurementSessionRestart):
(WKPageSetPrivateClickMeasurementTokenPublicKeyURLForTesting):
(WKPageSetPrivateClickMeasurementTokenSignatureURLForTesting):
(WKPageSetPrivateClickMeasurementAttributionReportURLsForTesting):
(WKPageMarkPrivateClickMeasurementsAsExpiredForTesting):
(WKPageSetPCMFraudPreventionValuesForTesting):
(WKPageSetPrivateClickMeasurementAppBundleIDForTesting):
(WKPageDispatchActivityStateUpdateForTesting):
(WKPageIsEditingCommandEnabledForTesting):
(WKPageSetPermissionLevelForTesting):
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _setStatisticsCrossSiteLoadWithLinkDecorationForTesting:withToHost:withWasFiltered:withCompletionHandler:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewTesting.mm:
(-[WKWebView _setDefersLoadingForTesting:]):
(-[WKWebView _setIndexOfGetDisplayMediaDeviceSelectedForTesting:]):
(-[WKWebView _setSystemCanPromptForGetDisplayMediaForTesting:]):
(-[WKWebView _setPrivateClickMeasurementOverrideTimerForTesting:completionHandler:]):
(-[WKWebView _setPrivateClickMeasurementAttributionReportURLsForTesting:destinationURL:completionHandler:]):
(-[WKWebView _setPrivateClickMeasurementAttributionTokenPublicKeyURLForTesting:completionHandler:]):
(-[WKWebView _setPrivateClickMeasurementAttributionTokenSignatureURLForTesting:completionHandler:]):
(-[WKWebView _setPrivateClickMeasurementAppBundleIDForTesting:completionHandler:]):
(-[WKWebView _dumpPrivateClickMeasurement:]):
(-[WKWebView _isLayerTreeFrozenForTesting:]):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::m_pageForTesting):
(WebKit::WebPageProxy::close):
(WebKit::WebPageProxy::clearNotificationPermissionState):
(WebKit::WebPageProxy::setCrossSiteLoadWithLinkDecorationForTesting): Deleted.
(WebKit::WebPageProxy::clearWheelEventTestMonitor): Deleted.
(WebKit::WebPageProxy::setDefersLoadingForTesting): Deleted.
(WebKit::WebPageProxy::dumpPrivateClickMeasurement): Deleted.
(WebKit::WebPageProxy::clearPrivateClickMeasurement): Deleted.
(WebKit::WebPageProxy::setPrivateClickMeasurementOverrideTimerForTesting): Deleted.
(WebKit::WebPageProxy::markAttributedPrivateClickMeasurementsAsExpiredForTesting): Deleted.
(WebKit::WebPageProxy::setPrivateClickMeasurementEphemeralMeasurementForTesting): Deleted.
(WebKit::WebPageProxy::simulatePrivateClickMeasurementSessionRestart): Deleted.
(WebKit::WebPageProxy::setPrivateClickMeasurementTokenPublicKeyURLForTesting): Deleted.
(WebKit::WebPageProxy::setPrivateClickMeasurementTokenSignatureURLForTesting): Deleted.
(WebKit::WebPageProxy::setPrivateClickMeasurementAttributionReportURLsForTesting): Deleted.
(WebKit::WebPageProxy::markPrivateClickMeasurementsAsExpiredForTesting): Deleted.
(WebKit::WebPageProxy::setPCMFraudPreventionValuesForTesting): Deleted.
(WebKit::WebPageProxy::setPrivateClickMeasurementAppBundleIDForTesting): Deleted.
(WebKit::WebPageProxy::dispatchActivityStateUpdateForTesting): Deleted.
(WebKit::WebPageProxy::isLayerTreeFrozen): Deleted.
(WebKit::WebPageProxy::setIndexOfGetDisplayMediaDeviceSelectedForTesting): Deleted.
(WebKit::WebPageProxy::setSystemCanPromptForGetDisplayMediaForTesting): Deleted.
(WebKit::WebPageProxy::isEditingCommandEnabledForTesting): Deleted.
(WebKit::WebPageProxy::setPermissionLevelForTesting): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxyTesting.cpp: Added.
(WebKit::WebPageProxyTesting::WebPageProxyTesting):
(WebKit::WebPageProxyTesting::sendMessage):
(WebKit::WebPageProxyTesting::sendMessageWithAsyncReply):
(WebKit::WebPageProxyTesting::messageSenderConnection const):
(WebKit::WebPageProxyTesting::messageSenderDestinationID const):
(WebKit::WebPageProxyTesting::setDefersLoading):
(WebKit::WebPageProxyTesting::dispatchActivityStateUpdate):
(WebKit::WebPageProxyTesting::isLayerTreeFrozen):
(WebKit::WebPageProxyTesting::setCrossSiteLoadWithLinkDecorationForTesting):
(WebKit::WebPageProxyTesting::setPermissionLevel):
(WebKit::WebPageProxyTesting::isEditingCommandEnabled):
(WebKit::WebPageProxyTesting::dumpPrivateClickMeasurement):
(WebKit::WebPageProxyTesting::clearPrivateClickMeasurement):
(WebKit::WebPageProxyTesting::setPrivateClickMeasurementOverrideTimer):
(WebKit::WebPageProxyTesting::markAttributedPrivateClickMeasurementsAsExpired):
(WebKit::WebPageProxyTesting::setPrivateClickMeasurementEphemeralMeasurement):
(WebKit::WebPageProxyTesting::simulatePrivateClickMeasurementSessionRestart):
(WebKit::WebPageProxyTesting::setPrivateClickMeasurementTokenPublicKeyURL):
(WebKit::WebPageProxyTesting::setPrivateClickMeasurementTokenSignatureURL):
(WebKit::WebPageProxyTesting::setPrivateClickMeasurementAttributionReportURLs):
(WebKit::WebPageProxyTesting::markPrivateClickMeasurementsAsExpired):
(WebKit::WebPageProxyTesting::setPCMFraudPreventionValues):
(WebKit::WebPageProxyTesting::setPrivateClickMeasurementAppBundleID):
(WebKit::WebPageProxyTesting::clearNotificationPermissionState):
(WebKit::WebPageProxyTesting::clearWheelEventTestMonitor):
(WebKit::WebPageProxyTesting::setIndexOfGetDisplayMediaDeviceSelectedForTesting):
(WebKit::WebPageProxyTesting::setSystemCanPromptForGetDisplayMediaForTesting):
* Source/WebKit/UIProcess/WebPageProxyTesting.h: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::WebPage):
(WebKit::WebPage::setBaseWritingDirection):
(WebKit::WebPage::close):
(WebKit::WebPage::isEditingCommandEnabled): Deleted.
(WebKit::WebPage::setDefersLoading): Deleted.
(WebKit::WebPage::isLayerTreeFrozen): Deleted.
(WebKit::WebPage::clearWheelEventTestMonitor): Deleted.
(WebKit::WebPage::clearNotificationPermissionState): Deleted.
(WebKit::WebPage::setPermissionLevelForTesting): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
(WebKit::WebPage::layerTreeFreezeReasons const):
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/WebPageTesting.cpp: Added.
(WebKit::WebPageTesting::WebPageTesting):
(WebKit::WebPageTesting::~WebPageTesting):
(WebKit::WebPageTesting::setDefersLoading):
(WebKit::WebPageTesting::isLayerTreeFrozen):
(WebKit::WebPageTesting::setPermissionLevel):
(WebKit::WebPageTesting::isEditingCommandEnabled):
(WebKit::WebPageTesting::clearNotificationPermissionState):
(WebKit::WebPageTesting::clearWheelEventTestMonitor):
* Source/WebKit/WebProcess/WebPage/WebPageTesting.h: Added.
* Source/WebKit/WebProcess/WebPage/WebPageTesting.messages.in: Added.

Canonical link: <a href="https://commits.webkit.org/280396@main">https://commits.webkit.org/280396@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a4beb1e31bc75a27e2861953fbd4e62c28172d3a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56505 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35831 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8977 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60112 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/6941 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43453 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7137 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/45781 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/59/builds/6941 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58534 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33693 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48763 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26634 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30470 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/5945 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/52450 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6361 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/61796 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/413 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/6474 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/53033 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/414 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48825 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/52951 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/367 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8393 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/31659 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32744 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/33827 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/32492 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -1571,6 +1574,8 @@ void WebPageProxy::close()
 
     m_provisionalPage = nullptr;
 
+    m_pageForTesting = nullptr;
```
Does this mean we need to null check it everywhere?

charliewolfe replied:
Probably, fixed.

charliewolfe committed:
Create `WebPageTesting` and `WebPageProxyTesting`
rdar://130555180
https://bugs.webkit.org/show_bug.cgi?id=275884

Reviewed by Alex Christensen.

WebPage and WebPageProxy have many functions that are intended to only be used for testing, this change
begins to move those functions into separate files. More testing-only functions will be added as we
create replacements for injected bundle API used by WKTR.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageClearWheelEventTestMonitor):
(WKPageDumpPrivateClickMeasurement):
(WKPageClearPrivateClickMeasurement):
(WKPageSetPrivateClickMeasurementOverrideTimerForTesting):
(WKPageMarkAttributedPrivateClickMeasurementsAsExpiredForTesting):
(WKPageSetPrivateClickMeasurementEphemeralMeasurementForTesting):
(WKPageSimulatePrivateClickMeasurementSessionRestart):
(WKPageSetPrivateClickMeasurementTokenPublicKeyURLForTesting):
(WKPageSetPrivateClickMeasurementTokenSignatureURLForTesting):
(WKPageSetPrivateClickMeasurementAttributionReportURLsForTesting):
(WKPageMarkPrivateClickMeasurementsAsExpiredForTesting):
(WKPageSetPCMFraudPreventionValuesForTesting):
(WKPageSetPrivateClickMeasurementAppBundleIDForTesting):
(WKPageDispatchActivityStateUpdateForTesting):
(WKPageIsEditingCommandEnabledForTesting):
(WKPageSetPermissionLevelForTesting):
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _setStatisticsCrossSiteLoadWithLinkDecorationForTesting:withToHost:withWasFiltered:withCompletionHandler:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewTesting.mm:
(-[WKWebView _setDefersLoadingForTesting:]):
(-[WKWebView _setIndexOfGetDisplayMediaDeviceSelectedForTesting:]):
(-[WKWebView _setSystemCanPromptForGetDisplayMediaForTesting:]):
(-[WKWebView _setPrivateClickMeasurementOverrideTimerForTesting:completionHandler:]):
(-[WKWebView _setPrivateClickMeasurementAttributionReportURLsForTesting:destinationURL:completionHandler:]):
(-[WKWebView _setPrivateClickMeasurementAttributionTokenPublicKeyURLForTesting:completionHandler:]):
(-[WKWebView _setPrivateClickMeasurementAttributionTokenSignatureURLForTesting:completionHandler:]):
(-[WKWebView _setPrivateClickMeasurementAppBundleIDForTesting:completionHandler:]):
(-[WKWebView _dumpPrivateClickMeasurement:]):
(-[WKWebView _isLayerTreeFrozenForTesting:]):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::m_pageForTesting):
(WebKit::WebPageProxy::close):
(WebKit::WebPageProxy::clearNotificationPermissionState):
(WebKit::WebPageProxy::setCrossSiteLoadWithLinkDecorationForTesting): Deleted.
(WebKit::WebPageProxy::clearWheelEventTestMonitor): Deleted.
(WebKit::WebPageProxy::setDefersLoadingForTesting): Deleted.
(WebKit::WebPageProxy::dumpPrivateClickMeasurement): Deleted.
(WebKit::WebPageProxy::clearPrivateClickMeasurement): Deleted.
(WebKit::WebPageProxy::setPrivateClickMeasurementOverrideTimerForTesting): Deleted.
(WebKit::WebPageProxy::markAttributedPrivateClickMeasurementsAsExpiredForTesting): Deleted.
(WebKit::WebPageProxy::setPrivateClickMeasurementEphemeralMeasurementForTesting): Deleted.
(WebKit::WebPageProxy::simulatePrivateClickMeasurementSessionRestart): Deleted.
(WebKit::WebPageProxy::setPrivateClickMeasurementTokenPublicKeyURLForTesting): Deleted.
(WebKit::WebPageProxy::setPrivateClickMeasurementTokenSignatureURLForTesting): Deleted.
(WebKit::WebPageProxy::setPrivateClickMeasurementAttributionReportURLsForTesting): Deleted.
(WebKit::WebPageProxy::markPrivateClickMeasurementsAsExpiredForTesting): Deleted.
(WebKit::WebPageProxy::setPCMFraudPreventionValuesForTesting): Deleted.
(WebKit::WebPageProxy::setPrivateClickMeasurementAppBundleIDForTesting): Deleted.
(WebKit::WebPageProxy::dispatchActivityStateUpdateForTesting): Deleted.
(WebKit::WebPageProxy::isLayerTreeFrozen): Deleted.
(WebKit::WebPageProxy::setIndexOfGetDisplayMediaDeviceSelectedForTesting): Deleted.
(WebKit::WebPageProxy::setSystemCanPromptForGetDisplayMediaForTesting): Deleted.
(WebKit::WebPageProxy::isEditingCommandEnabledForTesting): Deleted.
(WebKit::WebPageProxy::setPermissionLevelForTesting): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxyTesting.cpp: Added.
(WebKit::WebPageProxyTesting::WebPageProxyTesting):
(WebKit::WebPageProxyTesting::sendMessage):
(WebKit::WebPageProxyTesting::sendMessageWithAsyncReply):
(WebKit::WebPageProxyTesting::messageSenderConnection const):
(WebKit::WebPageProxyTesting::messageSenderDestinationID const):
(WebKit::WebPageProxyTesting::setDefersLoading):
(WebKit::WebPageProxyTesting::dispatchActivityStateUpdate):
(WebKit::WebPageProxyTesting::isLayerTreeFrozen):
(WebKit::WebPageProxyTesting::setCrossSiteLoadWithLinkDecorationForTesting):
(WebKit::WebPageProxyTesting::setPermissionLevel):
(WebKit::WebPageProxyTesting::isEditingCommandEnabled):
(WebKit::WebPageProxyTesting::dumpPrivateClickMeasurement):
(WebKit::WebPageProxyTesting::clearPrivateClickMeasurement):
(WebKit::WebPageProxyTesting::setPrivateClickMeasurementOverrideTimer):
(WebKit::WebPageProxyTesting::markAttributedPrivateClickMeasurementsAsExpired):
(WebKit::WebPageProxyTesting::setPrivateClickMeasurementEphemeralMeasurement):
(WebKit::WebPageProxyTesting::simulatePrivateClickMeasurementSessionRestart):
(WebKit::WebPageProxyTesting::setPrivateClickMeasurementTokenPublicKeyURL):
(WebKit::WebPageProxyTesting::setPrivateClickMeasurementTokenSignatureURL):
(WebKit::WebPageProxyTesting::setPrivateClickMeasurementAttributionReportURLs):
(WebKit::WebPageProxyTesting::markPrivateClickMeasurementsAsExpired):
(WebKit::WebPageProxyTesting::setPCMFraudPreventionValues):
(WebKit::WebPageProxyTesting::setPrivateClickMeasurementAppBundleID):
(WebKit::WebPageProxyTesting::clearNotificationPermissionState):
(WebKit::WebPageProxyTesting::clearWheelEventTestMonitor):
(WebKit::WebPageProxyTesting::setIndexOfGetDisplayMediaDeviceSelectedForTesting):
(WebKit::WebPageProxyTesting::setSystemCanPromptForGetDisplayMediaForTesting):
* Source/WebKit/UIProcess/WebPageProxyTesting.h: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::WebPage):
(WebKit::WebPage::setBaseWritingDirection):
(WebKit::WebPage::close):
(WebKit::WebPage::isEditingCommandEnabled): Deleted.
(WebKit::WebPage::setDefersLoading): Deleted.
(WebKit::WebPage::isLayerTreeFrozen): Deleted.
(WebKit::WebPage::clearWheelEventTestMonitor): Deleted.
(WebKit::WebPage::clearNotificationPermissionState): Deleted.
(WebKit::WebPage::setPermissionLevelForTesting): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
(WebKit::WebPage::layerTreeFreezeReasons const):
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/WebPageTesting.cpp: Added.
(WebKit::WebPageTesting::WebPageTesting):
(WebKit::WebPageTesting::~WebPageTesting):
(WebKit::WebPageTesting::setDefersLoading):
(WebKit::WebPageTesting::isLayerTreeFrozen):
(WebKit::WebPageTesting::setPermissionLevel):
(WebKit::WebPageTesting::isEditingCommandEnabled):
(WebKit::WebPageTesting::clearNotificationPermissionState):
(WebKit::WebPageTesting::clearWheelEventTestMonitor):
* Source/WebKit/WebProcess/WebPage/WebPageTesting.h: Added.
* Source/WebKit/WebProcess/WebPage/WebPageTesting.messages.in: Added.

Canonical link: https://commits.webkit.org/280396@main
